hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSEditLogAsync.java

```diff
@@ -63,6 +68,9 @@
             DFS_NAMENODE_EDITS_ASYNC_LOGGING_PENDING_QUEUE_SIZE_DEFAULT);
 
     editPendingQ = new ArrayBlockingQueue<>(editPendingQSize);
+
+    // the thread pool size should be configurable later, and justified with a rationale
+    logSyncNotifyExecutor = Executors.newFixedThreadPool(10);
```

Comment:
I prefer to make this improvement be more configurable to use. Can we make thread pool size be configurable in this PR? if the pool size is configured as 0, that means this improvements is disabled.

Reply:
Sure. What should be the default value? Many users use the default value, so probably we shouldn't set it as 0 by default.

Reply:
@functioner , we could make 10 as the default pool size.